### ISSUE TITLE
web: add field base url to login form

### ISSUE THREAD

**Issue by scanf** [HiOA-ABI/nikita-noark5-core#40]
This field should only be visible if the `$scope.base_url` is not already set. Having a base url would make switching servers for testing easier and avoid hardcoding.

- [ ] Use environment variable to determine if we already have a base url.
- [ ] Add field to login.html

**Comment by tsodring** [comment]
I have added  baseUrl field so that it appears in the loginPage.html. Let me know if this is enough. I'm not sure how to check for  $scope.base_url

**Comment by tsodring** [comment]
I think I can close this issue as this [issue](https://github.com/HiOA-ABI/nikita-noark5-core/issues/45) overrides this issue. I will work on a better general approach.

**Comment by scanf** [comment]
@tsodring This issue was created as a reminder for myself, looks like you are misunderstanding it. Please reopen and assign it to me. This is not the same as #45

**Comment by petterreinholdtsen** [comment]
There is now a nikitaOptions.baseUrl setting in web/javascripts/config.js for this.  Setting it from the web interface remains.

**Comment by oleaamot** [comment]
There is a bug when I login to a Nikita Noark 5 Core server (linux) on http://linux:3000/
from a remote web client (portable).

It seems like the web interface returns localhost:8092, while it should connect to
the baseUrl specified in nikita-noark5-core/web/dependencies/internal/config.js

I have this baseUrl in nikita-noark5-core/web/dependencies/internal/config.js on
the server (linux):

...
nikitaOptions = {
    //baseUrl: "http://n5test.kxml.no/api",
    baseUrl: 'http://linux:8092/noark5v4/',
    guiBaseUrl: 'http://linux:3000/',
    appUrl: 'http://linux:8092/noark5v4/hateoas-api',
    loginUrl: "http://linux:8092/noark5v4/auth",
    protocol: 'http',
    appName: 'noark5v4',
    apiName: 'hateoas-api',
    authPoint: 'auth',
    displayFooterNote: true,
    displayBreadcrumb: true,
    enabled: true
};
...

I get this error message on the web client (portable) when it connects to the Nikita Noark 5 Core server on http://linux:3000/

Could not find series object to retrieve related files!{"data":null,"status":-1,"config":{"method":"GET","transformRequest":[null],"transformResponse":[null],"jsonpCallbackParam":"callback","url":"http://localhost:8092/noark5v4/hateoas-api/arkivstruktur","headers":{"Authorization":"eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImF1ZGllbmNlIjoid2ViIiwiY3JlYXRlZCI6MTQ5OTE2MjUzNDA5OSwiZXhwIjoxNDk5NzY3MzM0fQ.9S-ahX6ZuIThBF1UiBPQDqM9Zeyu0J9z8FQKwV8Q5M5OD1CC8-AhsOIQUHnhouYhmQ1cuMc_us-IvgJTdbefDQ","Accept":"application/json, text/plain, */*"}},"statusText":""}

**Comment by tsodring** [comment]
Hi Ole,
I think this might relate to nikita-core, nit nikita-gui. There is a property to be set in nikita that governs the urls nikita uses. Take a look in nikita-noark5-core/core-webapp/src/main/resources. There you will find a file  called application-demo.yml. Open the file, and locate the line ` publicAddress: http://localhost:8092/noark5v4`. In your case I guess you want this to be ` publicAddress: http://linux:8092/noark5v4`. Restart nikita and check if the GUI is working now. 

This change was introduced after we put nikita behind apache. We no longer could use the incoming address to determine the outgoing addresses as the forwarding from apache meant all incoming requests came from localhost.

Give it a try and let me know if it works ...

**Comment by oleaamot** [comment]
Yes, I was able to login to the GUI from the remote web client after modifying the line `publicAddress: http://localhost:8092/noark5v4` in nikita-noark5-core/core-webapp/src/main/resources/application-demo.yml and restarting the Nikita Noark 5 Core server.

Thank you.
